discovery-provider/src/tasks/entity_manager/utils.py

```diff
@@ -171,6 +172,7 @@ def __init__(
         self.txhash = txhash
         self.new_records = new_records
         self.existing_records = existing_records
+        self.logger = logger
```

Comment:
Maybe add a comment here saying this is a logger with the event args as context (so people know the dif between this and simply importing StructuredLogger)